File: src/main/java/uz/com/journal/app/attendance/AttendanceService.java
package uz.com.journal.app.attendance;


import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import uz.com.journal.core.NonDeletableService;
import uz.com.platform.app.users.User;

public interface AttendanceService extends NonDeletableService<Attendance> {
    Page<Attendance> findPage(AttendanceFetch fetch, AttendanceCriteria criteria, Pageable pageable, User currentUser);

    Attendance findOne(String uuid, AttendanceFetch fetch, User currentUser);

    Attendance create(AttendanceItem item, User currentUser);

    Attendance update(String uuid, AttendanceItem item, User currentUser);
}

File: src/main/webapp/WEB-INF/src/js/app/administration/attendeds/AttendanceService.js
module.exports = function ($http) {
    var _this = this;

    _this.findPage = function () {
        var params = {
            size: 10000000,
            page: 0
        };
        return $http.get('/api/attendances', {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.findOne = function (uuid) {
        var params = {};
        return $http.get('/api/attendances/' + uuid, {params: params}).then(function (response) {
            return response.data;
        });
    };

    _this.create = function (attendance) {
        return $http.post('/api/attendances', attendance).then(function (response) {
            return response.data;
        });
    };

    _this.update = function (attendance) {
        return $http.put('/api/attendances/' + attendance.uuid, attendance).then(function (response) {
            return response.data;
        })
    };

    _this.delete = function (attendance) {
        return $http.delete('/api/attendances/' + attendance.uuid);
    }
};